src/lib/validators/openApiDiff.ts

```diff
@@ -186,52 +229,46 @@ export class OpenApiDiff {
         throw new Error('newSwagger is a required parameter of type "string" and it cannot be an empty string.')
     }
 
-    log.debug(`oldSwagger = "${oldSwagger}"`);
-    log.debug(`newSwagger = "${newSwagger}"`);
-
-    let OpenApiDiffPromise = new Promise<unknown>((resolve, reject) => {
-      if (!fs.existsSync(oldSwagger)) {
-        reject(`File "${oldSwagger}" not found.`);
-      }
+    log.debug(`oldSwagger = "${oldSwagger}"`)
+    log.debug(`newSwagger = "${newSwagger}"`)
 
-      if (!fs.existsSync(newSwagger)) {
-        reject(`File "${newSwagger}" not found.`);
-      }
+    if (!fs.existsSync(oldSwagger)) {
+      throw new Error(`File "${oldSwagger}" not found.`)
+    }
 
-      let cmd = `${self.dotNetPath()} ${self.openApiDiffDllPath()} -o ${oldSwagger} -n ${newSwagger}`;
-      if (self.options.json)
-      {
-        cmd = `${cmd} -JsonValidationMessages`;
-      }
+    if (!fs.existsSync(newSwagger)) {
+      throw new Error(`File "${newSwagger}" not found.`)
+    }
 
-      log.debug(`Executing: "${cmd}"`);
-      exec(cmd, { encoding: 'utf8', maxBuffer: 1024 * 1024 * 64 }, (err: unknown, stdout: unknown, stderr: unknown) => {
-        if (err) {
-          reject(err);
-        }
+    const cmd = `${this.dotNetPath()} ${this.openApiDiffDllPath()} -o ${oldSwagger} -n ${newSwagger}`
 
-        resolve(stdout);
-      });
-    });
+    log.debug(`Executing: "${cmd}"`)
+    const { stdout } = await exec(cmd, { encoding: 'utf8', maxBuffer: 1024 * 1024 * 64 })
+    const json = jsonParser.parse("", stdout) as Messages
 
-    return OpenApiDiffPromise;
+    const newJson = json.map(v => ({
+      ...v,
+      new: updateChange(v.new, newSwaggerFile),
+      old: updateChange(v.old, oldSwaggerFile),
+    }))
+    return JSON.stringify(newJson)
   }
 }
 
 // Testing
-// let swagger = '/Users/vishrut/git-repos/azure-rest-api-specs/arm-network/2017-03-01/swagger/virtualNetworkGateway.json';
-// let obj = new OpenApiDiff();
+// let swagger = '/Users/vishrut/git-repos/azure-rest-api-specs/arm-network/2017-03-01/swagger/virtualNetworkGateway.json'
```

Comment:
is this section useful? or should it be removed?

Reply:
Right. We don't need it anymore because we have unit tests now.